packages/hardhat-core/src/internal/cli/autocomplete.ts

```diff
@@ -1,28 +1,59 @@
+import findup from "find-up";
+import * as fs from "fs-extra";
+import * as path from "path";
+
 import { HARDHAT_PARAM_DEFINITIONS } from "../core/params/hardhat-params";
 import type hardhat from "../lib/hardhat-lib";
+import { getCacheDir } from "../util/global-dir";
 
 import { ArgumentsParser } from "./ArgumentsParser";
 
+type GlobalParam = keyof typeof HARDHAT_PARAM_DEFINITIONS;
+
 interface CompletionEnv {
   line: string;
   point: number;
 }
 
+interface CompletionData {
+  networks: string[];
+  tasks: {
+    [taskName: string]: {
+      name: string;
+      paramDefinitions: {
+        [paramName: string]: {
+          name: string;
```

Comment:
is the name stored twice? don't we need to store the type too? or maybe this is the typename?

Reply:
I think the name is stored twice. Not sure though.

We aren't using the type right now, but maybe we'll do in the future? So yeah, maybe it's a good idea to add it to this interface.